org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/extensions/general/completion/FilePathCompletionParticipant.java

```diff
@@ -0,0 +1,265 @@
+		int completionOffset = request.getOffset(); // offset after the typed character
+		int parsedAttributeStartOffset = StringUtils.getOffsetAfterWhitespace(fullAttributeValue, completionOffset - documentStartOffset) + documentStartOffset; // first character of URI
+		String subAttributeValue = text.substring(parsedAttributeStartOffset, completionOffset);
+	
+		Position startValue = xmlDocument.positionAt(parsedAttributeStartOffset);
+		Position endValue = xmlDocument.positionAt(completionOffset);
+		fullRange = new Range(startValue, endValue);
+
+		// Try to get the URI string from the attribute value in case it has a file scheme
+		// header (eg: "file://")
+		String pathInAttribute = subAttributeValue;
+		boolean hasFileScheme = false;
+
+		hasFileScheme = subAttributeValue.startsWith(FilesUtils.FILE_SCHEME);
+		if (hasFileScheme) {
+			pathInAttribute = subAttributeValue.substring(FilesUtils.FILE_SCHEME.length());
+		}
+
+		String slash = getFilePathSlash(subAttributeValue);
+
+		if (hasFileScheme) {
+			if (!pathInAttribute.startsWith("/")) {
+				return; // use of 'file://' and the path was not absolute
+			}
+			if (isWindows && pathInAttribute.length() == 1) { // only '/', so list Windows Drives
+				
+				Range replaceRange = adjustReplaceRange(xmlDocument, fullRange, subAttributeValue, "/");
+
+				File[] drives = File.listRoots();
+				for (File drive : drives) {
+					createFilePathCompletionItem(Paths.get("/"), drive, replaceRange, response, "/");
+				}
+				return;
+			}
+		}
+		else {
+			if(isWindows && "/".equals(slash)) { // non-URI using forward slash in Windows
+				return;
+			}
+		}
+		
+		if(isWindows) {
+			pathInAttribute = cleanPathForWindows(pathInAttribute);
+		}
+		
+		// Get the normalized URI string from the parent directory file if necessary
+		String directoryOfFile = null;
+
+		if (!hasFileScheme) { //pathInAttribute is not a uri, so we might need to reference the directory path
+			String uriString = xmlDocument.getTextDocument().getUri();
+			URI uri = URI.create(uriString);
+
+			if(!FILE_SCHEME.equals(uri.getScheme())) {
+				return;
+			}
+
+			String uriPathString = uri.getPath();
+			if(!uriPathString.startsWith("/")) {
+				return; //file uri is incorrect
+			}
+			int lastSlash = uriPathString.lastIndexOf("/");
+			if(lastSlash > -1) {
+				directoryOfFile = uriPathString.substring(0, lastSlash);
+
+				if(isWindows) {
+					// Necessary to that this path is readable in Windows
+					directoryOfFile = cleanPathForWindows(directoryOfFile);
+				}
+			}
+		}
+		
+		//Try to get an correctly formatted path from the given values
+		Path validAttributeValuePath = getNormalizedPath(directoryOfFile, pathInAttribute); 
+		
+		if(validAttributeValuePath == null) {
+			return;
+		}
+
+		//Get adjusted range for the completion item (insert at end, or overwrite some existing text in the path)
+		Range replaceRange = adjustReplaceRange(xmlDocument, fullRange, subAttributeValue, slash);
```

Comment:
I think replace range should be fullRange every time to update the whole attribute value (like completion based on XML Schema).

Reply:
I think it makes it harder, if you have 
```xml
<a k="AAA BBB ./fileA/| CCC"> </a>
```
If you try to replace the whole attribute then you have to parse it anyways to find out what values you don't touch ("AAA BBB ", " CCC")

Reply:
I understand but is it the commons case? If i take noschemanamzspace location attribute i prefer it replaces the attribute value. What do you think Fred?

Reply:
@NikolasKomonen I understand more your strategy now. We could see that in an another issue.